src/dataflow/src/source/postgres.rs

```diff
@@ -269,33 +304,27 @@ impl PostgresSourceReader {
                             .try_into()
                             .expect("software more than 200k years old, consider updating");
 
-                        // Here we report that we've seen everything up until `self.lsn`, which is
-                        // the lsn of that commit message we received. However, there could be
-                        // transactions that we have not seen yet (perhaps because they are not
-                        // committed yet) that started way earlier than self.lsn. Therefore, it's
-                        // unclear how to calculate a safe lower bound value to report back to the
-                        // server. The answer to this question will allow writing retry logic for
-                        // the replication phase.
-                        stream
-                            .as_mut()
-                            .standby_status_update(self.lsn, self.lsn, self.lsn, ts, 0)
-                            .await?;
+                        try_recoverable!(
```

Comment:
Are we no longer concerned about incorrectly reporting the wrong lower bound lsn?

Reply:
No! I looked up how postgres handles these messages and it does the right thing. When we send an LSN `x` as a standby message postgres calculates the LSN of the BEGIN statement of the earliest transaction that commits after `x`, so we're not going to lose any data.

Similarly, when we request to start replication from position `x` postgres does the same calculation and starts enough in the past so that no transaction is left behind.